# <center>Ejecutar código en notebooks a través de sockets</center>

En este ejemplo abrimos un canal de socket desde este notebook. Está configurado para ejecutar el código que reciba, a no ser que sean las palabras "off" o "hola" (y variantes), que sirven para apagar el socket servidor y lanzar el primer mensaje al canal. 

In [1]:
PORT = 8765

from server import echo_server
echo_server(PORT)

Esperando a recibir clientes...
Esperando a recibir mensajes del cliente...
> Recibido: b'hola'
< Enviando (127.0.0.1:41696): b'hola'
Esperando a recibir mensajes del cliente...
> Recibido: b'print(9999)'
< Enviando (127.0.0.1:41696): b'print(9999)'
9999
print(9999)
Esperando a recibir mensajes del cliente...
> Recibido: b'off'
< Enviando (127.0.0.1:41696): b'off'
Apagando el servidor...


Para ejecutar el cliente, dentro de esta misma carpeta desde una consola podemos ejecutar: 

<center>`python3 client.py --port 8765 --message "print(9999)"`</center>